flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/CatalogLoader.java

```diff
@@ -155,4 +210,55 @@ public String toString() {
     }
   }
 
+  class BaseCatalogLoader implements CatalogLoader {
+
+    private final Map<String, String> options;
+    private final String catalogName;
+    private final SerializableConfiguration hadoopConf;
+
+    private BaseCatalogLoader(
+            CatalogFactory.Context context
+    ) {
+      this.options = Maps.newHashMap(context.getOptions()); // wrap into a hashmap for serialization
+      this.catalogName = context.getName();
+      // Check to see if there are hadoop classes loaded in Flink's application classloader
+      if (isHadoopEnv(context.getClassLoader().getParent())) {
```

Comment:
The intention here is to only check Flink's application-level class loader (i.e. jars from flink's `/lib` folder).  This is important because the call retrieve the hadoop conf from the flink environment (`HadoopUtils.getHadoopConfiguration(GlobalConfiguration.loadConfiguration())`) does not have access to user level classes.

Reply:
We also have an interface marker, `org.apache.iceberg.hadoop.Configurable<C>`, which is used for a similar purpose (to avoid the dependency on hadoop).

It can be mixed in and then there's a marker you can check possibly. Here's an exmaple usage of it in `GlueCatalog`: https://github.com/apache/iceberg/blob/8afcdffd905fa284f20f6f84fba649d28fb7d923/aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java#L76-L82

Reply:
What is the use of this Interface marker ??
I still get the error , I  am using flink 1.13 and  
```
<dependency>
			<groupId>org.apache.iceberg</groupId>
			<artifactId>iceberg-flink-runtime-1.13</artifactId>
			<version>0.13.1</version>
		</dependency>
```